diff --git a/compat/siphash/siphash.c b/compat/siphash/siphash.c
index 7dc72cb..60249b8 100644
--- a/compat/siphash/siphash.c
+++ b/compat/siphash/siphash.c
@@ -26,6 +26,12 @@
 #include <asm/word-at-a-time.h>
 #endif
 
+#if __has_attribute(__fallthrough__)
+# define fallthrough                    __attribute__((__fallthrough__))
+#else
+# define fallthrough                    do {} while (0)  /* fallthrough */
+#endif
+
 #define SIPROUND \
 	do { \
 	v0 += v1; v1 = rol64(v1, 13); v1 ^= v0; v0 = rol64(v0, 32); \
